cln_lspd.go

```diff
@@ -0,0 +1,281 @@
+
+	capacity := incomingAmountMsat/1000 + additionalChannelCapacity
+	if capacity == publicChannelAmount {
+		capacity++
+	}
+	minConf = 0 //need to be updated with mindepth for zeroconf
+
+	//open private channel
+	channelPoint, err := clientcln.FundChannelExt(destination, glightning.NewSat(int(capacity)), nil, false, &minConf, nil)
+	if err != nil {
+		log.Printf("clientcln.OpenChannelSync(%v, %v) error: %v", destination, capacity, err)
+		return "", 0, err
+	}
+	return channelPoint.FundingTxId, uint32(0), err
+}
+
+func clnIsConnected(clientcln *glightning.Lightning, destination string) error {
+	pubKey := destination
+	peers, err := clientcln.ListPeers()
+	if err != nil {
+		//log.Printf("clientcln.ListPeers() error: %v", err)
+		return fmt.Errorf("clientcln.ListPeers() error: %w", err)
+	}
+	for _, peer := range peers {
+		if pubKey == peer.Id {
+			log.Printf("destination online: %v", destination)
+			return nil
+		}
+	}
+	log.Printf("destination offline: %v", destination)
+	return fmt.Errorf("destination offline")
+}
+func OnHtlcAccepted(event *glightning.HtlcAcceptedEvent) (*glightning.HtlcAcceptedResponse, error) {
+	log.Printf("htlc_accepted called\n")
+	onion := event.Onion
+
+	log.Printf("htlc: %v\nchanID: %v\nincoming amount: %v\noutgoing amount: %v\nincoming expiry: %v\noutgoing expiry: %v\npaymentHash: %v\nonionBlob: %v\n\n",
+		event.Htlc,
+		onion.ShortChannelId,
+		event.Htlc.AmountMilliSatoshi, //with fees
+		onion.ForwardAmount,
+		event.Htlc.CltvExpiryRelative,
+		event.Htlc.CltvExpiry,
+		event.Htlc.PaymentHash,
+		onion,
+	)
+
+	paymentHash, paymentSecret, destination, incomingAmountMsat, outgoingAmountMsat, fundingTxID, fundingTxOutnum, err := paymentInfo([]byte(event.Htlc.PaymentHash))
+	if err != nil {
+		log.Printf("paymentInfo(%v)\nfundingTxOutnum: %v\n error: %v", event.Htlc.PaymentHash, fundingTxOutnum, err)
+	}
+	var zeroConfFundingTxID string
+	//var zeroConfFundingTxOutnum uint32
+	if hex.EncodeToString(paymentSecret) != "" {
+		if hex.EncodeToString(fundingTxID) == "" {
+			if hex.EncodeToString(paymentHash) == event.Htlc.PaymentHash {
+
+				zeroConfFundingTxID, _, err = clnOpenChannel(clientcln, hex.EncodeToString(paymentHash), hex.EncodeToString(destination), incomingAmountMsat)
+				log.Printf("openclnOpenChannelChannel(%v, %v) err: %v", destination, incomingAmountMsat, err)
+				if err != nil {
+					log.Printf(" clnOpenChannel error: %v", err)
+				}
+
+			} else { //probing
+				failureCode := "TEMPORARY_CHANNEL_FAILURE"
+				if err := clnIsConnected(clientcln, hex.EncodeToString(destination)); err == nil {
+					failureCode = "INCORRECT_OR_UNKNOWN_PAYMENT_DETAILS"
+				}
+				log.Printf(" failureCode %v , Error %v", failureCode, err)
+			}
+
+			return clnResumeOrCancel(clientcln, hex.EncodeToString(destination), zeroConfFundingTxID, hex.EncodeToString(paymentHash), uint64(outgoingAmountMsat), 99, event)
```

Comment:
I see we are not running here in a separate go routing and this is good as long as the plugin hook function is called in a separate go routing for each interception. Is that the case?

Reply:
yes, on every attempt and every htlc intercept it is called again.